clippy_lints/src/utils/mod.rs

```diff
@@ -50,17 +50,77 @@ use rustc_lint::{LateContext, Level, Lint, LintContext};
 use rustc_middle::hir::map::Map;
 use rustc_middle::ty::subst::{GenericArg, GenericArgKind};
 use rustc_middle::ty::{self, layout::IntegerExt, Ty, TyCtxt, TypeFoldable};
+use rustc_session::Session;
 use rustc_span::hygiene::{ExpnKind, MacroKind};
 use rustc_span::source_map::original_sp;
 use rustc_span::sym as rustc_sym;
 use rustc_span::symbol::{self, kw, Symbol};
 use rustc_span::{BytePos, Pos, Span, DUMMY_SP};
 use rustc_target::abi::Integer;
 use rustc_trait_selection::traits::query::normalize::AtExt;
+use semver::Version;
 use smallvec::SmallVec;
 
 use crate::consts::{constant, Constant};
 
+pub fn parse_msrv(msrv: Option<String>, sess: Option<&Session>, span: Option<Span>) -> Option<Version> {
+    match msrv {
+        Some(val) => match Version::parse(&val) {
+            Ok(version) => Some(version),
+            Err(_) => {
+                if sess.is_some() && span.is_some() {
+                    sess.unwrap()
+                        .span_err(span.unwrap(), &format!("`{}` is not a valid Rust version!", val));
```

Comment:
```suggestion
                        .span_err(span.unwrap(), &format!("`{}` is not a valid Rust version", val));
```

It isn't allowed to be excited about errors ;)